### PR TITLE
Feature/leave auction

### PR DESCRIPTION
**Pull request by dmitriy-uvin**

### CODE REVIEW

File: packages/frontend/components/item/item-info-auction/component.tsx [hunk 1]
```diff
@@ -79,6 +97,20 @@ export const ItemInfoAuction = ({
           ></FavoriteButton>
         </div>
       </form>
+      {!!isAbleToLeaveAuction && user && (
+        <div css={styles.leaveAuctionBlock}>
+          <Button
+            onClick={onLeaveAuction}
+            variant="danger"
+            tooltip="You will be excluded from the auction and all your bids are deleted!"
```
Comment (Nightlngale) [inline packages/frontend/components/item/item-info-auction/component.tsx:105]: localisation

**Review by Nightlngale — CHANGES_REQUESTED** [review]

File: packages/backend/src/repositories/bid.ts [hunk 2]
```diff
@@ -17,4 +17,22 @@ export class BidRepository {
       },
     });
   }
+
+  async getByUserAndProduct(userId: string, productId: string) {
+    return await this._dbClient.bid.findMany({
```
Comment (Nightlngale) [inline packages/backend/src/repositories/bid.ts:22]: why return await

File: packages/backend/src/repositories/bid.ts [hunk 3]
```diff
@@ -17,4 +17,22 @@ export class BidRepository {
...
+      where: {
+        bidderId: userId,
+        productId: productId,
+      },
+    });
+  }
+
+  async deleteAllByProductAndUser(userId: string, productId: string) {
+    return await this._dbClient.bid.deleteMany({
```
Comment (Nightlngale) [inline packages/backend/src/repositories/bid.ts:31]: why return await

File: packages/backend/src/services/product.ts [hunk 4]
```diff
@@ -71,6 +76,42 @@ export class ProductService {
...
+      throw new ProductNotFoundError();
+    }
+
+    if (toUtc(product.endDate) < toUtc()) {
+      throw new AuctionEndedError();
+    }
+
+    await this._bidRepository.deleteAllByProductAndUser(userId, productId);
+
+    return await this.getById(productId);
```
Comment (Nightlngale) [inline packages/backend/src/services/product.ts:112]: why return await

**Review by Nightlngale — CHANGES_REQUESTED** [review]

File: packages/backend/src/repositories/bid.ts (hunk 2 above)
Comment (dmitriy-uvin) [reply packages/backend/src/repositories/bid.ts:22]: because prisma returns Promise

**Review by Nightlngale — APPROVED** [review]

**Pull request merged**